### ISSUE TITLE
[19w24a] Stretched half-block cannot be interacted with for part of its length

### ISSUE THREAD

**Issue by Dwarf-LordPangolin** [tsunamayo/Skywanderers#1306]
### Short Description of the issue:
When the half-block is stretched out, the last meter and a half of it cannot be interacted with.

### Steps to reproduce the issue:
Place a half-block that is stretched out so that it is more than one and a half meters deep. The last meter and a half cannot be interacted with (deleted, painted, have stuff placed on it, etc).

### Environment
- Skywanderers version: 19w24a
- Graphics Card: NVIDIA GeForce GTX 1060 6GB

### Additional information:
![20190701005159_1](https://user-images.githubusercontent.com/33041786/60413465-a3fc4a80-9b9a-11e9-9f75-96b0d5d90e04.jpg)

[output_log.txt](https://github.com/tsunamayo/Skywanderers/files/3344100/output_log.txt)

**Comment by tsunamayo** [comment]
Okay, actually this block is only supposed to stretch lateraly - I  fixed that, so there is no issues. 
Btw do you think this block is usefull? What kind of use do you have for it? I want to remove it, to force players to build interior with full block as there is quite a few issue with light linked to them. Cheers

**Comment by ProPeach** [comment]
The half blocks are definitely useful for all kinds of situations, they're an integral block shape in my opinion. Can you expand more on what you mean bu players building interiors in full blocks? Lots of an interiors are built with bricks too

**Comment by Dwarf-LordPangolin** [comment]
They're definitely useful. It would be possible to build without them, but we'd end up using a lot more bricks.

**Comment by tsunamayo** [comment]
I was referring to block used for build wall and floor. 
Using brick for that is not a good idea as brick dont stop light, and as we will now enter into that phase, they wont provide heat dissipation and hull armor value.
I think having that form factor was more a leftover from minecraft heritage with the slab, but now I think it is a bit at odd with the rest of the shapes selection, I dont really see the point, and it can cause a bit of confusion.

**Comment by ProPeach** [comment]
Ok, thanks for clarifying.
When I talked about bricks being used for the interior, I meant that the bricks are used for decorating a box made of blocks on medium/large ships. This makes the half slab even more useful as you can get more interior space, as you have on your own Corellian shuttle. The block also has unlimited uses in other parts of the ship too, it's definitely not redundant.
In regards to bricks no contributing to heat dissipation or armour value, how are fighter sized ships like Sen's Goblin going to work? I put forward a suggestion to combat brick spam in this thread (#1233), I think it addresses your concerns well and works with gameplay too.

**Comment by ultrak2k** [comment]
Whoa hold on, bricks not contributing to anything and being cosmetic would killl... everything wouldn’t it?

**Comment by ZachZent** [comment]
Tsuna, I have a question. When was the last time you have built a ship? Like not just gone in a modified your old... really old shuttle? When was the last time you have seen what we have been building since the graphics update? Because unless I am misinterpreting what you are saying, you believe that blocks make up most people's ships, both in and out. That has never been the case in a long time. Ever since we got stretchy bricks.  If you look at a lot of people's builds, you will see how much people rely on bricks and other small parts. From small fighters to even the largest of ships, every single one uses bricks in their own way. You will never see a ship made only or majority out of blocks, especially when you get to the interior. These are not for decoration either. These are functioning bricks doing a block's job. 

I am linking an imgur album full of ships, interiors and exteriors people have built ever since we got the graphics update.
https://imgur.com/a/P0ZsY3W

Maybe you didn't intend bricks to function in this role, but that is what you got. This isn't bad at all, the level of creativity here is astounding. Something that no other game allows. If you turn bricks fully cosmetic, all of the ships you see in the album and many more that were not included would become useless. Sky Wanderers players would be forced to choose between function and looks. Do you know what that results in? Starmade.
https://cdn.discordapp.com/attachments/271723356795961345/559521930617683981/starmade-screenshot-0055.png
This type of ship will be the only way to get down to the smaller size with any sort of detail. If it means we don't get 200 ships in one "instance" but still get the full creativity of bricks, I would be more the happy.

Again, if I am misinterpreting this then I apologize. Bricks shouldn't be 1:1 effective in mass/heat/etc as blocks, but removing them from actual gameplay will result in many players going elsewhere

**Comment by Majikmonster** [comment]
Wait, If bricks don't contribute to function of SOME kind, then wouldn't that make them an additional liability? Repairing them would take resources, using them would take space, and keeping them would create fluff that weighs down a craft. I don't see how that would result in anything good for people who want more than Starmade or Minecraft. If well-working ships are achieved through blocks alone, then bricks will be a running joke in the game, as anyone concerned with the functionality or sustainability of their builds will toss out bricks as just extra mass.

**Comment by tsunamayo** [comment]
Thanks, yes I havent build a ship for a long time, I want to start a new one soon.
Those ships looks amazing, are they available on the discord? Now that I am getting a lot more into gameplay I am gonna need your help to balance system and things like that.

For small ship I am not sure yet, I have a few lead to explore. It will likely be a set of rules that allows some freedom for smaller build, giving them a small "boost", boost being not significant for larger build.
Obviously I dont want to "kill brick", I mean I worked my ass to put them in the game, it would have been so so much easier to just stick to voxel ;)

Interior looked mostly legit (maybe not the tiled floor, but I will give you stickers for that), except maybe when too cramped but for another reason:  I want to put crew / ship boarding one day , and they are definitely not gonna like interiors that are too cramped. Also for crew I will need to standardize a lot of things for pathfinding ect (like door), but I think it is totally worth it - I mean wouldnt you like to hire a crew of a 100s, having to build them quarter, board and fight inside enemy ship, crew having a rarity system (some legendary crew would have special skills), and levelling up... I definitely want to that.

But an interior should be at least 3m high (honestly 2m high is way too claustrophobic), so it seems reasonable to expect player to use block for that. If we are talking interior room and corridor, we are talking medium / big ship already, so fighter are not really concerned. Again the biggest reason is lighting: brick dont stop light, light leaks everywhere and its ugly... 
Yes I used those in the corellian shuttle, and thats why I want to remove them after seeing all the crappy lighting situation they create! Honestly this shuttle interior is too cramped...

Also if I where to give some kind of hull value, then I guess player would find empty voxel inside their ship and stack them with bricks. Or just use brick inside of voxel as that would give them an advantage. If you say then give a lower armor density compared to block hull to avoid this, then it means that the value will be very very tiny, so a ship with 0.25 cm hull or wings would have a really negligeable armor value. And again as my whole repair / damage / armor system will be per voxel for various reason having only hull block contribute to hull is making much more sense from a code and logic point of view.

For repair / damage, everything is gonna be per voxel, including repair cost, so you dont have to worry about that - you will repair all the brick contained in a voxel at once. Also I wanted to have free repair in shipyard, only in-combat repair would cost some resources (for gameplay balance reason).

But nothing is set in stone at this point, we can continue to discuss further, I will look after your build and it will most likely change quite a bit at the beginning when I balance combat.
Cheers!
